core/src/models/batch_id.rs

```diff
@@ -46,6 +48,10 @@ impl BatchId {
         self.order_collection_start_time() + BATCH_DURATION
     }
 
+    pub fn solve_end_time(self) -> SystemTime {
+        self.solve_start_time() + SOLVING_WINDOW
+    }
```

Comment:
What happens if `self.solve_start_time() + SOLVING_WINDOW` is greater than the batch length? Do we have anything checking or imposing such a constraint?

Reply:
The constants for batch duration and solving window are defined right next to each other with one clearly being larger than the other so I don't think there is a need to assert this more.

Reply:
Rust needs `static_assert!`s :stuck_out_tongue: